### PR TITLE
Episodes

### PR DESCRIPTION
**Pull request by Dbrantley**
What I have for the TV show class. If anything else needs to be added please let me know

### CODE REVIEW

**Review by joeyfreeman — CHANGES_REQUESTED** [review]
Does not compile. Multiple syntax errors.

**Comment by Dbrantley** [comment]
Alrighty will go iron those out

File: PlotTalk/Src/tvshow.h [hunk 1]
```diff
@@ -9,18 +9,21 @@
...
+    GetSeason(SeasonId);
+
 
-    //setConnection(Connection& connection);
 
 private:
-    //Connection connection;
-};
+
+//json Connector};
```
Comment (joeyfreeman) [inline PlotTalk/Src/tvshow.h:25]: Closing } commented out, so class definition doesn't compile

File: PlotTalk/Src/episode.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#endif // EPISODE_H
+
+#include<Qstring>
+#include<QVector>
+
+class Episode
+{
+protected:
+    Episode(); //constructor
+    Episode(int inMediaID, Qstring inUserName, Qstring inText);
```
Comment (joeyfreeman) [inline PlotTalk/Src/episode.h]: Constructors should be public. Only inherited classes could instantiate Episodes if set to protected.

File: PlotTalk/Src/episode.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+seasonId="";
+EpSummary="";
+}
+
+Episode::Episode(int inMediaID, Qstring summary, Qstring inText)
+{
+
+    this->TvshowID=mediaId;
+    this->TvTitle=EpisodeTitle;
+    this->Summary=EpSummary;
```
Comment (joeyfreeman) [inline PlotTalk/Src/episode.cpp]: Qstring should be QString, TvshowID, TvTitle and Summary are not part of Episode

File: PlotTalk/Src/tvshow.h [hunk 4]
```diff
@@ -9,18 +9,21 @@
...
+    QString Link;
     QString graphicLink;
-
+    int SeasonId; //Foreign Key For Seasons Class
     // TODO: Add Seasons once Season class is available.
     TvShow();
-    TvShow(QString name, QString tmdbLink, QString graphicLink);
+    TvShow(int showId,QString Title, QString Link, QString graphicLink);
+    GetTvShow();
+    GetSeason(SeasonId);
```
Comment (joeyfreeman) [inline PlotTalk/Src/tvshow.h]: Can't use class variable name (SeasonId) in function declaration like this

Comment (Dbrantley) [reply PlotTalk/Src/tvshow.h]: Hmmm so got episode class mixed up with tvshow and also some syntax errors! Sorry Will repush once i fix everything

**Review by allanhaywood — CHANGES_REQUESTED** [review]
Please add unit tests for this new class and any changes made.

In addition, update any existing relevant tests.

File: PlotTalk/Src/episode.h [hunk 5]
```diff
@@ -0,0 +1,29 @@
+#ifndef EPISODE_H
+#define EPISODE_H
+
+#endif // EPISODE_H
+
+#include<Qstring>
```
Comment (joeyfreeman) [inline PlotTalk/Src/episode.h]: Qstring

File: PlotTalk/Src/tvshow.h [hunk 6]
```diff
@@ -5,22 +5,24 @@
...
     TvShow(QString name, QString tmdbLink, QString graphicLink);
 
-    //setConnection(Connection& connection);
-
-private:
-    //Connection connection;
 };
 
-#endif // TVSHOW_H
+#endif
```
Comment (allanhaywood) [inline PlotTalk/Src/tvshow.h:26]: It's good practice to label the endif here, in this case it would have // TVSHOW_H

**Review by allanhaywood — CHANGES_REQUESTED** [review]
There is a test failure. I believe we made this change on our skype call, perhaps you didn't do a git add on the changes.

onfig: Using QtTest library 5.7.0, Qt 5.7.0 (i386-little_endian-ilp32 shared (dynamic) debug build; by MSVC 2015)
PASS   : TestTvShow::initTestCase()
FAIL!  : TestTvShow::TestConstructor() Compared values are not the same
   Actual   (ShowId)       : 0
   Expected (tvShow.showId): 9434012
..\PlotTalk\Test\testtvshow.cpp(19) : failure location
PASS   : TestTvShow::cleanupTestCase()
Totals: 2 passed, 1 failed, 0 skipped, 0 blacklisted, 1ms

**Review by allanhaywood — APPROVED** [review]

**Review by joeyfreeman — APPROVED** [review]
Updated from master and moved Review and Comment vectors to episode.h. Compiles and passes all tests on my machine.

**Pull request merged**